include/unifex/detach_on_cancel.hpp

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include <unifex/get_stop_token.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <atomic>
+#include <exception>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _detach_on_cancel {
+
+template <typename UpstreamSender, typename DownstreamReceiver>
+struct operation_state {
+  struct _receiver;
+
+  struct detached_state;
+
+  struct cancel_callback final {
+    detached_state* state_;
+
+    void operator()() noexcept {
+      auto* op = state_->parentOp_.exchange(nullptr);
+      if (op != nullptr) {
+        // We won the race for cancellation
+        op->state_->stopSource_.request_stop();
+        // ownership is transferred to the detached state
+        (void)op->state_.release();
+        op->callback_.destruct();
+        unifex::set_done(std::move(op->receiver_));
+      }
+    }
+  };
+
+  struct type final {
+    type(UpstreamSender&& s, DownstreamReceiver&& r)
+      : receiver_(std::move(r))
+      , state_(new detached_state(this, (UpstreamSender &&) s)){};
+
+    friend auto tag_invoke(tag_t<unifex::start>, typename operation_state<UpstreamSender, DownstreamReceiver>::type& op) noexcept {
```

Comment:
`start()` always returns `void` so could use `void` return-type here instead of `auto`.